src/Symfony/Component/Messenger/Transport/Sender/SendersLocator.php

```diff
@@ -56,4 +55,27 @@ public function getSenders(Envelope $envelope, ?bool &$handle = false): iterable
 
         $handle = $handle || null === $sender;
     }
+
+    /**
+     * @internal
+     */
+    protected function listTypes(Envelope $envelope): array
```

Comment:
this should be private

Reply:
Also `\Symfony\Component\Messenger\Handler\HandlersLocator::listTypes` would not need to be public anymore when not used here.